tool/preprocess/match.go

```diff
@@ -37,17 +37,44 @@ import (
 type ruleMatcher struct {
 	availableRules map[string][]rules.InstRule
 	moduleVersions []*vendorModule // vendor used only
+	projectDeps    map[string]bool // actual dependencies from dry run commands
 }
 
-func newRuleMatcher() *ruleMatcher {
-	rules := make(map[string][]rules.InstRule)
+func newRuleMatcher(compileCmds []string) *ruleMatcher {
+	availableRules := make(map[string][]rules.InstRule)
 	for _, rule := range findAvailableRules() {
-		rules[rule.GetImportPath()] = append(rules[rule.GetImportPath()], rule)
+		availableRules[rule.GetImportPath()] = append(availableRules[rule.GetImportPath()], rule)
 	}
 	if config.GetConf().Verbose {
-		util.Log("Available rules: %v", rules)
+		util.Log("Available rules: %v", availableRules)
+	}
+
+	// Populated projectDeps from compileCmds
+	projectDeps := populateDependenciesFromCmd(compileCmds)
+
+	return &ruleMatcher{
+		availableRules: availableRules,
+		projectDeps:    projectDeps,
 	}
-	return &ruleMatcher{availableRules: rules}
+}
+
+// populateDependenciesFromCmd extracts import paths from the compile commands
+func populateDependenciesFromCmd(compileCmds []string) map[string]bool {
+	projectDeps := make(map[string]bool)
+
+	for _, cmd := range compileCmds {
+		cmdArgs := util.SplitCmds(cmd)
+		importPath := findFlagValue(cmdArgs, util.BuildPattern)
+		if importPath != "" {
```

Comment:
Logically, importPath should never be empty. I suggest using an assert to confirm this assumption.

```
importPath := findFlagValue(cmdArgs, util.BuildPattern)
util.Assert(importPath!="","sanity check")
projectDeps[importPath] = true
```